### PR TITLE
[process-agent] Create unit tests for process check

### PR DESCRIPTION
**Pull request by robertjli**
### What does this PR do?

Add unit tests for the process check

### Motivation

Increase test coverage

### Describe how to test/QA your changes

N/A

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] ~Changed code has automated tests for its functionality.~
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] ~If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).~
- [ ] ~If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.~
- [ ] ~If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.~
- [ ] ~If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.~

### CODE REVIEW

File: pkg/process/checks/process_test.go [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+}
+
+func TestProcessCheckWithRealtime(t *testing.T) {
+	processCheck, probe := processCheckWithMockProbe(t)
+
+	proc1 := makeProcess(1, "git clone google.com")
+	proc2 := makeProcess(2, "mine-bitcoins -all -x")
+	proc3 := makeProcess(3, "foo --version")
+	proc4 := makeProcess(4, "foo -bar -bim")
+	proc5 := makeProcess(5, "datadog-process-agent -ddconfig datadog.conf")
```
Comment (just-chillin) [inline pkg/process/checks/process_test.go:190]: nit: `-ddconfig` is deprecated, so we shouldn't reference it in our code, use `--cfgpath` instead.

**Review by just-chillin — APPROVED** [review]

File: pkg/process/checks/process_test.go (hunk 1 above)
Comment (robertjli) [reply pkg/process/checks/process_test.go]: Done!

**Review by xornivore — COMMENTED** [review]
Looks great, should we also add tests for `StatsForPIDs` (which is used by RT check)? We only test `ProcessesByPID` here.

**Comment by robertjli** [comment]
> Looks great, should we also add tests for StatsForPIDs (which is used by RT check)? We only test ProcessesByPID here.

This is tested in https://github.com/DataDog/datadog-agent/pull/12976

**Pull request merged**